tests/test_search.py

```diff
@@ -4,12 +4,261 @@
 from util.solver_workflow import new_solver_session  # noqa: F401
 
 import ansys.fluent.core as pyfluent
-from ansys.fluent.core.utils.search import _get_version_path_prefix_from_obj
+from ansys.fluent.core.utils.search import (
+    _get_api_object_names,
+    _get_capitalize_match_for_word_from_names,
+    _get_close_matches_for_word_from_names,
+    _get_exact_match_for_word_from_names,
+    _get_match_case_for_word_from_names,
+    _get_version_path_prefix_from_obj,
+    _get_wildcard_matches_for_word_from_names,
+    _search,
+    _search_semantic,
+    _search_whole_word,
+    _search_wildcard,
+)
+
+
+@pytest.mark.fluent_version("==24.2")
+def test_nltk_data_download():
+    import nltk
+
+    packages = ["wordnet", "omw-1.4"]
+    for package in packages:
+        nltk.download(package, quiet=True)
+
+    _search_semantic("读", language="cmn")
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_exact_match_for_word_from_names():
+    names = _get_api_object_names()
+    exact_match = _get_exact_match_for_word_from_names("VideoResoutionY", names)
+    assert "VideoResoutionY" in exact_match
+    assert len(exact_match) == 1
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_capitalize_match_for_word_from_names():
+    names = _get_api_object_names()
+    capitalize_match_cases = _get_capitalize_match_for_word_from_names("font", names)
+    assert "font" not in capitalize_match_cases
+    assert "Font" in capitalize_match_cases
+    assert "TextFontAutomaticHorizontalSize" in capitalize_match_cases
+    assert "TextFontAutomaticUnits" in capitalize_match_cases
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_match_case_for_word_from_names():
+    names = _get_api_object_names()
+    match_cases = _get_match_case_for_word_from_names("font", names)
+    for match_case in match_cases:
+        assert "Font" not in match_case
+        assert "font" in match_case
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_wildcard_matches_for_word_from_names():
+    names = _get_api_object_names()
+    wildcard_matches = _get_wildcard_matches_for_word_from_names("iter*", names)
+    assert "iterating" in wildcard_matches
+    assert "iterate_steady_2way_fsi" in wildcard_matches
```

Comment:
If we precisisely define names in the test, we have much more control. We assert exactly what the full result should be, and we can specify various sets of names to suit given test cases.

Reply:
Ok. We will also check exact full result.